### PR TITLE
Change no data page logic according to core changes

### PR DESCRIPTION
**Pull request by sgiehl**
### Description:

With https://github.com/matomo-org/matomo/pull/21029 the handling of no data page will be fully refactored. This PR includes the requires changes for this plugin to work with the new core code.

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

**Comment by sgiehl** [comment]
@AltamashShaikh We have completely refactored how the no data screen works in Matomo. The changes here will adjust this plugin, so it show the tab as before. The changes will be part of Matomo 5.0.0-rc3, which was not yet released. Once that was released we might need to release a new version of this plugin with the changes in that PR as well.

File: SiteContentDetection/GoogleAnalyticsImporter.php [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+use Piwik\Plugins\SitesManager\SiteContentDetection\GoogleAnalytics3;
+use Piwik\Plugins\SitesManager\SiteContentDetection\GoogleAnalytics4;
+use Piwik\SiteContentDetector;
+use Piwik\View;
+
+class GoogleAnalyticsImporter extends \Piwik\Plugins\SitesManager\SiteContentDetection\SiteContentDetectionAbstract
+{
+    public static function getName(): string
+    {
+        return 'Google Analytics Import';
```
Comment (AltamashShaikh) [inline SiteContentDetection/GoogleAnalyticsImporter.php]:
```suggestion
        return Piwik::translate('GoogleAnalyticsImporter_AdminMenuTitle');
```

File: tests/Fixtures/EmptySiteWithSiteContentDetectionGA.php [hunk 2]
```diff
@@ -22,21 +25,9 @@ class EmptySiteWithSiteContentDetectionGA extends Fixture
...
-            'ga4' => false,
-            'gtm' => false,
-            'cloudflare' => true,
-            'cms' => SitesManager::SITE_TYPE_UNKNOWN
-        ];
-
         return [
             SiteContentDetector::class => \Piwik\DI::autowire(FakeSiteContentDetector::class)
-                ->constructorParameter('mockData', $mockData)
+                ->constructorParameter('detectedContentDetections', [Cloudflare::getId(), GoogleAnalytics3::getId()])
```
Comment (AltamashShaikh) [inline tests/Fixtures/EmptySiteWithSiteContentDetectionGA.php:28]: @sgiehl Similar changes would be required in ConenctAccounts and Cloud plugin

**Review by AltamashShaikh — APPROVED** [review]
Looks good and everything works as expected, left just 1 change request for translation rest all looks good

File: tests/Fixtures/EmptySiteWithSiteContentDetectionGA.php [hunk 3]
```diff
@@ -7,6 +7,9 @@
  */
 namespace Piwik\Plugins\GoogleAnalyticsImporter\tests\Fixtures;
 
+use Piwik\Plugins\SEO\Metric\Google;
```
Comment (AltamashShaikh) [inline tests/Fixtures/EmptySiteWithSiteContentDetectionGA.php]: @sgiehl This use is required?

**Comment by AltamashShaikh** [comment]
@sgiehl Can you also update this [line](https://github.com/matomo-org/plugin-GoogleAnalyticsImporter/blob/5.x-dev/Controller.php#L316) in this PR as `ga-import` is changed to `googleanalyticsimporter`

File: SiteContentDetection/GoogleAnalyticsImporter.php (hunk 1 above)
Comment (sgiehl) [reply SiteContentDetection/GoogleAnalyticsImporter.php]: Looks like I forgot to commit that, as I had that change done locally already 🙈

**Comment by sgiehl** [comment]
@AltamashShaikh applied your feedback

**Review by AltamashShaikh — APPROVED** [review]

**Pull request merged**